services/todo-legacy/controllers/todo-controller.ts

```diff
@@ -0,0 +1,35 @@
+import { api, Application, inject } from '@loopback/core';
+import { def } from './todo-controller.api';
+import { TodoRepository } from '../repositories/todo';
+import * as util from 'util';
+
+@api(def)
+export class TodoController {
+  @inject('repositories.todo')
+  repository: TodoRepository;
+
+  constructor() {
+  }
+
+  async getTodo(id?: number, filter?: any) {
```

Comment:
The loopback-next injection will satisfy the parameters in matching order based on their ordering in the API definition file. So, in this case, my id and filter are ordered as such and I can use the filter param _without_ id to retrieve items by filter. This allows me to recycle the same handler logic for both use cases at the controller-level.